sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubAuthenticationPolicy.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http.Headers;
+using System.Text;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Messaging.WebPubSub
+{
+    /// <summary>
+    /// Web PubSub Authentication Policy.
+    /// </summary>
+    internal partial class WebPubSubAuthenticationPolicy : HttpPipelineSynchronousPolicy
+    {
+        private readonly AzureKeyCredential _credential;
+
+        /// <summary>
+        /// Creates an instance of the authentication policy
+        /// </summary>
+        /// <param name="credential"></param>
+        public WebPubSubAuthenticationPolicy(AzureKeyCredential credential) => _credential = credential;
+
+        /// <inheritdoc/>
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            string audience = message.Request.Uri.ToUri().ToString();
+            string accessToken = JwtUtils.GenerateJwtBearer(audience, claims: null, expiresAfter: TimeSpan.FromMinutes(10), _credential);
```

Comment:
Where does that magical 10 minutes come from?  Shouldn't we at least calculate the maximum length possible with all retries + max jitter on construction of this policy?

Reply:
I remember having a discussion about this and all languages decided to settle on 10 mins. @vicancy, do you know the reason for 10 mins?

Reply:
we will have more discussion about these timeouts the first beta. I will capture these in an issue.